activities/blocking-and-triggers/CustomTrigger.cs

```diff
@@ -0,0 +1,425 @@
+    /// </summary>
+    [Input(
+        DisplayName = "Correlation Value",
+        Description = "Optional correlation value to match against the signal (e.g., order ID, user ID)"
+    )]
+    public Input<string?> CorrelationValue { get; set; } = default!;
+
+    /// <summary>
+    /// Whether this trigger can start new workflow instances.
+    /// Set to true if you want this signal to create new workflows.
+    /// </summary>
+    [Input(
+        DisplayName = "Can Start Workflow",
+        Description = "Whether receiving this signal can start a new workflow instance",
+        DefaultValue = false
+    )]
+    public Input<bool> CanStartWorkflow { get; set; } = new(false);
+
+    /// <summary>
+    /// Output containing the signal payload received when the workflow resumes.
+    /// </summary>
+    [Output(
+        DisplayName = "Signal Payload",
+        Description = "The data payload received with the signal"
+    )]
+    public Output<object?> SignalPayload { get; set; } = default!;
+
+    /// <inheritdoc />
+    protected override async ValueTask ExecuteAsync(ActivityExecutionContext context)
+    {
+        // Get the signal name and correlation value from inputs
+        var signalName = context.Get(SignalName);
+        var correlationValue = context.Get(CorrelationValue);
+
+        // Check if we're resuming from a bookmark (signal was received)
+        if (context.GetIsBookmarkResumed())
+        {
+            // Signal was received, extract the payload
+            // NOTE: The exact API for getting bookmark input may vary by Elsa v3 version
+            var payload = context.GetWorkflowInput("SignalPayload");
+            context.Set(SignalPayload, payload);
+            
+            await context.CompleteActivityAsync();
+            return;
+        }
+
+        // Create a bookmark to wait for the signal
+        // NOTE: Bookmark creation API may vary by Elsa v3 version
+        var bookmarkPayload = new CustomSignalBookmark
+        {
+            SignalName = signalName!,
+            CorrelationValue = correlationValue
+        };
+
+        // Create the bookmark with a unique name
+        var bookmarkName = $"CustomSignal:{signalName}";
+        if (!string.IsNullOrEmpty(correlationValue))
+        {
+            bookmarkName += $":{correlationValue}";
+        }
+
+        // NOTE: The CreateBookmark method signature may vary by Elsa v3 version
+        // Some versions may use different parameters or options
+        context.CreateBookmark(
+            bookmarkPayload, 
+            callback: OnResumeAsync,
+            includeActivityInstanceId: false);
+
+        // The workflow will suspend here and wait for the signal
+        // No need to explicitly call Suspend() - it's implicit when creating a bookmark
+    }
+
+    /// <summary>
+    /// Called when the bookmark is resumed (signal received).
+    /// NOTE: This callback pattern may vary by Elsa v3 version.
```

Comment:
This callback pattern is consistent across Elsa 3 versions. Please remove this part of the comment.